### PR TITLE
Make sure to have a consistent_group_by_blockID

### PR DESCRIPTION
**Pull request by Natim**
Today, it is possible that the blockID is once i1222 and once i1223 randomly, this aims to make sure we will always have i1222

### CODE REVIEW

**Comment by EnTeQuAk** [comment]
Looks good to me, I'm just missing some tests that are able to verify it works as expected. Mind adding something? I don't remember any flappy tests that randomly fail so it didn't occur in our test-suite.

**Comment by Natim** [comment]
Yes I will thanks :)

**Comment by EnTeQuAk** [comment]
Hmm, seems you broke all the tests.

**Comment by Natim** [comment]
Yes I am working on it :)
I will ask for review when it will be ready :)

**Comment by Natim** [comment]
Ok there is a failling test to explain what I am trying to fix.

**Comment by Natim** [comment]
They are great :)

File: src/olympia/blocklist/tests/test_views.py [hunk 1]
```diff
@@ -192,6 +192,32 @@ def test_block_id(self):
         item = self.dom(self.fx4_url).getElementsByTagName('emItem')[0]
         assert item.getAttribute('blockID') == 'i' + str(self.details.id)
 
+    def test_block_id_consistency(self):
+        details = BlocklistDetail.objects.create(
+            name="blocked item",
```
Comment (EnTeQuAk) [inline src/olympia/blocklist/tests/test_views.py]: Can we use single quotes as we do everywhere else and not mix them?

**Comment by EnTeQuAk** [comment]
eh, yeah, sometimes I start reviewing stuff even if I'm not supposed to :laughing: 

r+wc

**Comment by Natim** [comment]
I don't think the last commit was supposed to make the tests fails...

**Comment by Natim** [comment]
Ok so we've got a random failing test on travis meaning that any of my solutions doesn't really fix it.

**Comment by EnTeQuAk** [comment]
@Natim there were some questions on Friday in IRC regarding what exactly this fixes and the actual fix probably should me somewhere around https://github.com/Natim/addons-server/blob/98a3b6eba6ad1dea58f875f7769ad10151f962fe/src/olympia/blocklist/views.py#L109-L122 since this block is essentially overwriting all the ordering we do earlier, thus ordering by `detail_id` doesn't really change anything.

**Comment by Natim** [comment]
Here are the logs about the discussion we had.

```
14:14:10 <eviljeff> natim: is there an issue the pull fixes?  I don't understand what it's fixing exactly
14:57:00 <natim> eviljeff: it is fixing random blockID in the XML blocklist file
14:59:46 <eviljeff> natim: what's a random blockID?  Assuming the blockID is the 'i456' number, its not random - it's the detail model's id/pk in the database. Can you include the bug/issue that explains it?
15:01:42 <eviljeff> i.e. https://addons.mozilla.org/firefox/blocked/i1231 is https://addons.mozilla.org/admin/models/blocklist/blocklistdetail/1231/ in django admin
15:03:17 <eviljeff> which is the id of https://github.com/mozilla/addons-server/blob/master/src/olympia/blocklist/models.py#L31 set when the block detail is added
15:07:13 <natim> eviljeff: let say that we have a group by that merge together versions of the same plugin with only one blockID even if each of them have got specific details. The current model is crappy reason why we are moving for a new one but in the mean time we would need to always have the smaller id so that we can have a consistent one
15:07:29 <natim> same addons not plugin sorry
15:08:23 <natim> We have a script that generates the same XML file as AMO does and we compare them
15:08:42 <natim> With a random ID we have randomly failing tests 
15:09:18 <natim> You can have multiple details for the same add-on guid
```

**Comment by Natim** [comment]
So basically we want to make sure that the group by ``blockID`` (``detail_id``) always uses the smaller one so that we can generate the same XML file (with the same blockID) using Kinto without using randomly one or the other.

**Comment by Natim** [comment]
After working this through, we now have two ways of fixing it.

Either like that https://github.com/mozilla/addons-server/pull/3053/commits/7240f84041e1c344275f612d2c04b1e011deedf3#diff-cc4082e617603c9321d8854ba43e0afeR103
or like that https://github.com/mozilla/addons-server/pull/3053/commits/c1d904948ae5256c68436a77b79708bd94a2feb2#diff-cc4082e617603c9321d8854ba43e0afeR120

@EnTeQuAk how do you prefer it to be?

**Comment by EnTeQuAk** [comment]
I still feel like doing something with the order_by doesn't really change anything since sorted_groupby modifies the order. So https://github.com/mozilla/addons-server/pull/3053/commits/f376e8ae9cfe4f03bdb9c0236e120dc4515ab8a3#diff-cc4082e617603c9321d8854ba43e0afeR120 looks more reasonable to me.

Also, just ignore CircleCI for a bit, it's optional and we're still figuring things out :)

**Comment by Natim** [comment]
Ok I will revert, do you want me to squash as well or do you rather prefer keep the history?

**Comment by EnTeQuAk** [comment]
I personally love to have the history but we always squash before merge into one or more reasonable commits.

**Comment by Natim** [comment]
Ok should I let you do that then?

**Comment by EnTeQuAk** [comment]
I just went ahead and did that :)

**Comment by Natim** [comment]
Ok great thanks for your patience @EnTeQuAk

**Comment by EnTeQuAk** [comment]
Sure, you did all the work so thank you :)

**Pull request merged**